lib/ASTGen/Sources/ASTGen/Literals.swift

```diff
@@ -11,18 +11,200 @@
 //===----------------------------------------------------------------------===//
 
 import ASTBridging
+@_spi(Compiler) import SwiftParser
 import SwiftSyntax
 
 extension ASTGenVisitor {
-  func generate(stringLiteralExpr node: StringLiteralExprSyntax) -> BridgedStringLiteralExpr {
-    let openDelimiterOrQuoteLoc = self.generateSourceLoc(node.openingPounds ?? node.openingQuote)
+  func generate(stringLiteralExpr node: StringLiteralExprSyntax) -> BridgedExpr {
+    if node.segments.allSatisfy({ $0.is(StringSegmentSyntax.self) }) {
+      return self.generateStringLiteralExpr(stringLiteralExpr: node).asExpr
+    } else {
+      return self.generateInterpolatedStringLiteralExpr(stringLiteralExpr: node).asExpr
+    }
+  }
 
-    // FIXME: Handle interpolated strings.
-    // FIXME: Avoid 'String' instantiation
-    var segment = node.segments.first!.as(StringSegmentSyntax.self)!.content.text
-    return segment.withBridgedString { bridgedSegment in
-      return .createParsed(self.ctx, value: bridgedSegment, loc: openDelimiterOrQuoteLoc)
+  func generateStringLiteralExpr(stringLiteralExpr node: StringLiteralExprSyntax) -> BridgedStringLiteralExpr {
```

Comment:
We should add a comment that this only handles the case where the string doesn’t contain any interpolation.

Reply:
Added assertion too.